hooks/post-turn-quality-stop-hook.py

```
@@ -0,0 +1,447 @@
+    changed_files: list[str] = field(default_factory=list)
+    categories: dict[str, bool] = field(default_factory=default_categories)
+    make_targets_requested: list[str] = field(default_factory=list)
+    make_targets_run: list[str] = field(default_factory=list)
+    make_targets_skipped: list[str] = field(default_factory=list)
+    commands: list[dict[str, Any]] = field(default_factory=list)
+    fetched: bool = False
+    error: str | None = None
+
+
+def run(cmd: list[str], cwd: Path) -> subprocess.CompletedProcess[str]:
+    # noqa: S603 - command and args are controlled (no shell, no user-supplied command strings).
```

Ruff (0.14.14)

[warning] Unused `noqa` directive (unused: `S603`)

Remove unused `noqa` directive

(RUF100)